core/trino-main/src/main/java/io/trino/operator/exchange/UniformPartitionRebalancer.java

```diff
@@ -162,17 +162,15 @@ private synchronized void rebalancePartitions(List<Long> writerPhysicalWrittenBy
             for (WriterId minSkewedWriter : minSkewedWriters) {
                 // There's no need to add the maxWriter back to priority queues if no partition rebalancing happened
                 if (context.rebalancePartition(maxWriter, minSkewedWriter)) {
```

Comment:
Could you just return `rebalancedAffectedWriters` from `rebalancePartition`? When it's empty it means rebalance didn't happen.